src/Style.hs

```diff
@@ -470,144 +469,73 @@ procObjFn :: (Autofloat a) =>
     Stmt ->
     [ObjFnInfo a]
 procObjFn varMap fns (ObjFn fname es) =
-    trStr ("New Objective function: " ++ fname ++ " " ++ (show names)) $
-    (func, defaultWeight, names, nums) : fns
+    trStr ("New Objective function: " ++ fname ++ " " ++ (show args)) $
+    (func, defaultWeight, args) : fns
     where
         func = case M.lookup fname objFuncDict of
             Just f -> f
             Nothing -> error ("procObjFn: objective function '" ++ fname ++ "' not known")
-        (names, nums) = partitionEithers $ map (procExpr varMap) es
+        args = map (procExpr varMap) es
 procObjFn varMap fns (Avoid fname es) = fns -- TODO: `avoid` functions
 procObjFn varMap fns _ = fns -- TODO: `avoid` functions
 
--- TODO: Have a more principled expr look up routine
 lookupVarMap :: String -> VarMap -> String
 lookupVarMap s varMap = case M.lookup s varMap of
     Just s' -> s'
-    Nothing -> case M.lookup s computationDict of -- TODO remove this case
-               Just f -> trace ("found function named: " ++ s) $ s
-               Nothing -> s
-               -- TODO: there is a possibility of accessing unselected Substance variables here. As written here, we are assuming all ids from Substance are accessible in Style GLOBALLY. Is this okay?
-               -- error $ "lookupVarMap: incorrect variable mapping from " ++ s ++ " or no computation"
+    Nothing -> s
+    -- HACK: allowing access to unselected Substance elements. NOT a principled solution
+    -- Nothing -> error $ "lookupVarMap: incorrect variable mapping from " ++ s ++ "; varMap content: \n" ++  show varMap
 
 -- | Resolve a Style expression, which could be operations among expressions such as a chained dot-access for an attribute through a couple of layers of indirection
-
--- An expression can be either a string (variable name) or float (literal)? TODO revise
 -- Example: context (VarMap) [X ~> A] would arise with Sub: "Set A", Sty: "Set X { ... }"
 -- TODO: this function is not done/tested yet
--- TODO: handle properties (e.g. `X.yaxis:length`, `X.yaxis.label:width`)
--- TODO: replace underscores with spaces
+-- TODO: handle properties (e.g. `X.yaxis.length`)
 -- TODO: generalize this function to return Exprs? (so objectives/constraints/computations can handle gets)
-procExpr :: (Autofloat a) => VarMap -> Expr -> Either String a
-
--- in context [X ~> A], look up "X", return "A"
--- TODO: or a list of all shapes: "A xaxis", "A yaxis", ... (lookupAll does this for now)
-procExpr ctx (Id subObjPattern) = Left $ lookupVarMap subObjPattern ctx
-
--- "ID.label" is a synonym for "ID.shape.label"
-procExpr ctx r@(BinOp Access (Id subObjName) (Id "label")) = error ("cannot access label of non-shape:\n" ++ show r)
-
+procExpr :: (Autofloat a) => VarMap -> Expr -> TypeIn a
+procExpr ctx (Id subObjPattern) = TAllShapes $ lookupVarMap subObjPattern ctx
+procExpr _ (StringLit s) = TStr s
+procExpr _ (IntLit i) = TNum $ r2f i -- TODO this shouldn't flatten ints for computations
```

Comment:
we have `TInt` now, so let's use that tag

Reply:
WIll propagate that change in Computations.  BTW what do you mean by `-- TODO this shouldn't flatten ints for computations`?